sql/core/src/main/scala/org/apache/spark/sql/execution/ExplainUtils.scala

```diff
@@ -148,9 +148,10 @@ object ExplainUtils extends AdaptiveSparkPlanHelper {
         setOpId(p)
       case other: QueryPlan[_] =>
         setOpId(other)
-        other.innerChildren.foldLeft(currentOperationID) {
+        currentOperationID = other.innerChildren.foldLeft(currentOperationID) {
           (curId, plan) => generateOperatorIDs(plan, curId)
         }
+        ()
```

Comment:
why do we need this?

Reply:
Removed. For some reason my intellij was complaining because it wasn't returning Unit, but it's fine without now.